user_profile.py

```diff
@@ -56,12 +56,181 @@
+        )
+        formData.add_argument(
+            "firstAddress",
+            type=str,
+            help="First address is required",
+            required=True,
+        )
+        formData.add_argument(
+            "secondAddress",
+            type=str,
+            help="Second address is required",
+            required=False,
+        )
+        formData.add_argument(
+            "city", type=str, help="City is required", required=True
+        )
+        formData.add_argument(
+            "province", type=str, help="Province is required", required=True
+        )
+        formData.add_argument(
+            "country", type=str, help="Country is required", required=True
+        )
+        formData.add_argument(
+            "postalCode",
+            type=str,
+            help="Postal Code is required and maximum of 10 characters long",
+            required=True,
+        )
+        formData.add_argument(
+            "gender", type=str, help="Gender is required", required=True
+        )
+        formData.add_argument(
+            "profile_image",
+            type=str,
+            help="This will be an image in bytes",
+            required=False,
+        )
+        formData.add_argument("religion", type=str, required=False)
+        formData.add_argument("user_bio", type=str, required=False)
+        formData.add_argument("user_interest", type=str, required=False)
+        formData.add_argument(
+            "education_institutions",
+            type=str,
+            help="University is required",
+            required=True,
+        )
+        formData.add_argument(
+            "education_majors",
+            type=str,
+            help="Majors are required",
+            required=True,
+        )
+        formData.add_argument(
+            "education_degrees",
+            type=str,
+            help="Degrees are required",
+            required=True,
+        )
+        formData.add_argument(
+            "graduation_date",
+            type=inputs.date,
+            help="Graduation Day is required",
+            required=True,
+        )
+        formData.add_argument(
+            "identification_option",
+            type=str,
+            help="User can choose which material they want to submit in order to verify their student's status",
+            required=True,
+        )
+        formData.add_argument(
+            "identification_material",
+            type=str,
+            help="User has to upload the material that verify their student's status",
+            required=True,
+        )
+
+    # a get method to get the user profile and return to the client
+    @token_required(permission_list=[
+        "can_view_dashboard",
+        "can_view_profile",
+        "can_change_profile"
+    ], secret_key=secret_key)
+    @marshal_with(userInformation_resource_fields)  # serialize the instance object
+    def get(self):
+        with user_profile_app.app_context():
+            if request.method == 'GET':
```

Comment:
Do you need to check `request.method` here? Doesn't the `Resource` class already handles this.

So if the method name is `get`, it only gets executed when request is `get`, same thing for `post`, `delete`, and `patch`